youtube_dl/downloader/hls.py

```diff
@@ -81,7 +89,10 @@ def real_download(self, filename, info_dict):
                     line
                     if re.match(r'^https?://', line)
                     else compat_urlparse.urljoin(man_url, line))
-                fragment_urls.append(segment_url)
+                if arrived_at_last_downloaded_segment:
+                    fragment_urls.append(segment_url)
```

Comment:
Since now you skip already downloaded segments the total fragment message displays wrong value after restarting: `[hlsnative] Total fragments: ...`.